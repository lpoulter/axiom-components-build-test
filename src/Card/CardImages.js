import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import CardImage from './CardImage';

export default class CardImages extends Component {
  static propTypes = {
    /** Class name to be appended to the element */
    className: PropTypes.string,
     /**
     * Aspect ratios for the height of the container depending on how many
     * images should be shown
     */
    ratios: PropTypes.array.isRequired,
    /** The multiple image sources to be used inside the image grid */
    srcs: PropTypes.arrayOf(PropTypes.string).isRequired,
  };

  static defaultProps = {
    ratios: [null, '50%', '75%', '100%'],
  };

  render() {
    const { className, ratios, srcs, ...rest } = this.props;
    const ratio = ratios[srcs.length - 1];
    const style = { paddingBottom: ratio };
    const classes = classnames('ax-card__images', `ax-card__images--${srcs.length}`, {
      'ax-card__images--ratio': ratio,
    }, className );

    return (
      <Base { ...rest } className={ classes } style={ style }>
        { ratio ? (
          <div className="ax-card__images-grid">
            { srcs.map((src, index) => (
              <div
                  className="ax-card__images-grid-item"
                  key={ index }
                  style={ { backgroundImage: `url(${src})` } } />
            )) }
          </div>
        ) : (
          <CardImage src={ srcs[0] } />
        ) }
      </Base>
    );
  }
}
