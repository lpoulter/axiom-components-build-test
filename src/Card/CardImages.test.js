import React from 'react';
import renderer from 'react-test-renderer';
import CardImages from './CardImages';

const getComponent = (props = {}) =>
  renderer.create(
    <CardImages { ...props } />
  );

describe('CardImages', () => {
  it('renders with 1 src', () => {
    const component = getComponent({
      srcs: ['image.png'],
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with 2 src', () => {
    const component = getComponent({
      srcs: ['image.png', 'image.png'],
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with 3 src', () => {
    const component = getComponent({
      srcs: ['image.png', 'image.png', 'image.png'],
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with 4 src', () => {
    const component = getComponent({
      srcs: ['image.png', 'image.png', 'image.png', 'image.png'],
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
