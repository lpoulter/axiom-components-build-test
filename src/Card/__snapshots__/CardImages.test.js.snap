// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImages renders with 1 src 1`] = `
<div
  className="ax-card__images ax-card__images--1"
  style={
    Object {
      "paddingBottom": null,
    }
  }
>
  <div
    className="ax-card__image"
  >
    <img
      className="ax-image ax-image--square ax-space--x0"
      src="image.png"
      style={
        Object {
          "backgroundColor": undefined,
          "height": undefined,
          "maxWidth": "100%",
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`CardImages renders with 2 src 1`] = `
<div
  className="ax-card__images ax-card__images--2 ax-card__images--ratio"
  style={
    Object {
      "paddingBottom": "50%",
    }
  }
>
  <div
    className="ax-card__images-grid"
  >
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
  </div>
</div>
`;

exports[`CardImages renders with 3 src 1`] = `
<div
  className="ax-card__images ax-card__images--3 ax-card__images--ratio"
  style={
    Object {
      "paddingBottom": "75%",
    }
  }
>
  <div
    className="ax-card__images-grid"
  >
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
  </div>
</div>
`;

exports[`CardImages renders with 4 src 1`] = `
<div
  className="ax-card__images ax-card__images--4 ax-card__images--ratio"
  style={
    Object {
      "paddingBottom": "100%",
    }
  }
>
  <div
    className="ax-card__images-grid"
  >
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
    <div
      className="ax-card__images-grid-item"
      style={
        Object {
          "backgroundImage": "url(image.png)",
        }
      }
    />
  </div>
</div>
`;
